Image-Factory/Factory.ps1

```diff
@@ -509,7 +571,18 @@ function RunTheFactory
             Copy-Item "$($ResourceDirectory)\bits" -Destination ($driveLetter + ":\") -Recurse;
             
             # Create first logon script
-            $updateCheckScriptBlock | Out-String | Out-File -FilePath "$($driveLetter):\Bits\Logon.ps1" -Width 4096;
+            if($UseStaticIP) {
+                $staticUpdateCheckScriptBlock = $updateCheckScriptBlock | Out-String
+                $staticUpdateCheckScriptBlock = $staticUpdateCheckScriptBlock.Replace('$UseStaticIP = $false', '$UseStaticIP = $true')
```

Comment:
Okay, I see what you are doing here.  However, this is confusing for someone reading the code.  Rather than having '$UseStaticIP = $false' and changing it if true - can we have '$UseStaticIP = STATICIPBOOLPLACEHOLDER' and then change it to true or false here depending on the input.  It will mean a little more code - but it will make it readable.